contracts/interfaces/IWitnetPriceRouter.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "./IERC165.sol";
+
+/// @title The Witnet Price Router basic interface.
+/// @dev Guides implementation of price feeds aggregation contracts.
+/// @author The Witnet Foundation.
+interface IWitnetPriceRouter
+{
+    /// Emitted everytime a price pair is attached to a new price feed contract
+    /// @dev See https://github.com/adoracles/ADOIPs/blob/main/adoip-0010.md 
+    /// @dev to learn how these ids are created.
+    event PricePairSet(bytes32 indexed erc2362ID, IERC165 pricefeed);
+
+    /// Returns the ERC-165-compliant price feed contract currently attending 
+    /// updates on the given price pair.
+    function getPriceFeed(bytes32 _erc2362id) external view returns (IERC165);
+
+    /// Returns human-readable ERC2362-based caption of the price pair being
+    /// attended by the given price feed contract address. 
+    /// @dev Should fail if the given price feed contract address is not currently
+    /// @dev supported by the router.
+    function getPriceFeedCaption(IERC165 _priceFeed) external view returns (string memory);
+
+    /// Helper pure function: returns hash of the provided ERC2362-compliant price pair caption.
+    function hashPriceCaption(string memory _priceCaption) external pure returns (bytes32);
```

Comment:
As per ERC2363, the hash of the caption is called ID.
```suggestion
    /// Helper pure function: returns hash of the provided ERC2362-compliant pair caption (aka ID).
    function pairId(string memory _pairCaption) external pure returns (bytes32);
```

Reply:
will call it `currencyPairId` for completeness, tough